eng/update-dependencies/DependencyUpdater.cs

```diff
@@ -0,0 +1,303 @@
+                            VersionGroupName = CabFileVersionGroupName,
+                            Regex = new Regex(@$"# Apply latest patch(.|\n)+dism.+C:\\patch\\(?<{CabFileVersionGroupName}>\S+)", RegexOptions.IgnoreCase)
+                        }
+                    }
+                );
+        }
+
+        private IEnumerable<IDependencyUpdater> CreateVsUpdaters()
+        {
+            VsConfig vsConfig = JsonConvert.DeserializeObject<VsConfig>(
+                File.ReadAllText(this.options.VsConfigPath));
+
+            const string TestAgentGroupName = "TestAgent";
+            const string BuildToolsGroupName = "BuildTools";
+            const string WebTargetsGroupName = "WebTargets";
+
+            return GetDockerfiles()
+                .SelectMany(dockerfile =>
+                    new IDependencyUpdater[]
+                    {
+                        new CustomFileRegexUpdater(vsConfig.TestAgentUrl, dockerfile.BuildInfoName)
+                        {
+                            VersionGroupName = TestAgentGroupName,
+                            Path = dockerfile.Path,
+                            Regex = new Regex(@$"(?<{TestAgentGroupName}>https:\/\/\S+vs_TestAgent\.exe)"),
+                        },
+                        new CustomFileRegexUpdater(vsConfig.BuildToolsUrl, dockerfile.BuildInfoName)
+                        {
+                            VersionGroupName = BuildToolsGroupName,
+                            Path = dockerfile.Path,
+                            Regex = new Regex(@$"(?<{BuildToolsGroupName}>https:\/\/\S+vs_BuildTools\.exe)"),
+                        },
+                        new CustomFileRegexUpdater(vsConfig.WebTargetsUrl, dockerfile.BuildInfoName)
+                        {
+                            VersionGroupName = WebTargetsGroupName,
+                            Path = dockerfile.Path,
+                            Regex = new Regex(@$"# Install web targets(.|\n)+?(?<{WebTargetsGroupName}>https:\/\/\S+)"),
+                        }
+                    });
+        }
+
+        private static string ParseCabFileName(string lcuDownloadUrl)
+        {
+            string msuFilename = lcuDownloadUrl.Substring(lcuDownloadUrl.LastIndexOf("/") + 1);
+            return msuFilename.Substring(0, msuFilename.IndexOf("_")) + ".cab";
+        }
+
+        private static (LcuConfig Config, string BuildInfoName)? GetLcuConfigInfo(DockerfileInfo dockerfile, LcuConfig[] lcuConfigs)
+        {
+            if (dockerfile.BuildInfoName != RuntimeBuildInfoName)
+            {
+                return null;
+            }
+
+            LcuConfig? lcuConfig = lcuConfigs
+                .FirstOrDefault(config => config.OsVersion == dockerfile.OsVersion &&
+                    config.RuntimeVersions.Any(runtime => runtime == dockerfile.FrameworkVersion));
+
+            if (lcuConfig != null)
+            {
+                return (lcuConfig, dockerfile.BuildInfoName);
```

Comment:
LCU only applies to the Runtime Dockerfiles.  Why does that need to be returned versus hard coding it in CreateLcuUpdaters?  Avoiding the anonymous type would help simplify the code.